src/category_theory/limits/has_limits.lean

```diff
@@ -506,6 +506,14 @@ instance : is_right_adjoint (lim : (J ⥤ C) ⥤ C) := ⟨_, const_lim_adj⟩
 
 end lim_functor
 
+instance lim_map_mono' {F G : J ⥤ C} [has_limits_of_shape J C] (α : F ⟶ G)
+  [mono α] : mono (lim_map α) :=
+(lim : (J ⥤ C) ⥤ C).map_mono α
+
+instance lim_map_mono {F G : J ⥤ C} [has_limit F] [has_limit G] (α : F ⟶ G)
+   [∀ j, mono (α.app j)] : mono (lim_map α) :=
+ ⟨λ Z u v h, limit.hom_ext $ λ j, (cancel_mono (α.app j)).1 $ by simpa using h =≫ limit.π _ j⟩
```

Comment:
I think both of versions are worth keeping, `lim_map_mono'` has weaker assumption at `mono α` and `lim_map_mono` doesn't require `has_limits_of_shape J C` @TwoFX